StreamChatClient_v3/Models/ChannelModel.swift

```diff
@@ -0,0 +1,258 @@
+    // MARK: - Internal
+    
+    /// A helper variable to cache the result of the filter for only banned members.
+    //  lazy var bannedMembers: Set<MemberModel<ExtraData.User>> = Set(self.members.filter { $0.isBanned })
+    
+    /// A list of users to invite in the channel.
+    let invitedMembers: Set<MemberModel<ExtraData.User>> // TODO: Why is this not public?
+    
+    internal init(id: String, extraData: ExtraData?, members: Set<UserModel<ExtraData.User>>) {
+        self.init(type: .unknown, id: ChannelId(type: .unknown, id: id))
+    }
+    
+    internal init(
+        type: ChannelType,
+        id: ChannelId,
+        lastMessageDate: Date? = nil,
+        created: Date = .init(),
+        updated: Date = .init(),
+        deleted: Date? = nil,
+        createdBy: UserModel<ExtraData.User>? = nil,
+        config: ChannelConfig = .init(),
+        frozen: Bool = false,
+        members: Set<MemberModel<ExtraData.User>> = [],
+        watchers: Set<UserModel<ExtraData.User>> = [],
+        team: String = "",
+        unreadCount: ChannelUnreadCount = .noUnread,
+        watcherCount: Int = 0,
+        unreadMessageRead: MessageRead<ExtraData.User>? = nil,
+        banEnabling: BanEnabling = .disabled,
+        isWatched: Bool = false,
+        extraData: ExtraData.Channel? = nil,
+        invitedMembers: Set<MemberModel<ExtraData.User>> = []
+    ) {
+        self.type = type
+        self.id = id
+        self.lastMessageDate = lastMessageDate
+        self.created = created
+        self.updated = updated
+        self.deleted = deleted
+        self.createdBy = createdBy
+        self.config = config
+        self.frozen = frozen
+        self.members = members
+        self.watchers = watchers
+        self.team = team
+        self.unreadCount = unreadCount
+        self.watcherCount = watcherCount
+        self.unreadMessageRead = unreadMessageRead
+        self.banEnabling = banEnabling
+        self.isWatched = isWatched
+        self.extraData = extraData
+        self.invitedMembers = invitedMembers
+    }
+}
+
+extension ChannelModel {
+    /// Check if the channel was deleted.
+    public var isDeleted: Bool { deleted != nil }
+    
+    /// Checks if read events evalable for the current user.
+    public var readEventsEnabled: Bool { /* config.readEventsEnabled && members.contains(Member.current) */ fatalError() }
+    
+    /// Checks if the channel is direct message type between 2 users.
+    public var isDirectMessage: Bool { id.id.hasPrefix("!members") && members.count == 2 }
+    
+    /// Checks if the current status of the channel is unread.
+    public var isUnread: Bool { unreadCount.messages > 0 }
+    
+    /// Checks for the channel data encoding is empty.
+    var isEmpty: Bool { /* extraData == nil && members.isEmpty && invitedMembers.isEmpty */ fatalError() }
+}
+
+/// A convenience `ChannelModel` typealias with no additional channel data.
+public typealias Channel = ChannelModel<DefaultDataTypes>
```

Comment:
One idea that could help to avoid specifying a custom extra data type for the user and etc:
```swift
public final class Client<ExtraData: ExtraDataTypes> {
    public typealias UserType = UserModel<ExtraData.User>
    public typealias ChannelType = ChannelModel<ExtraData.Channel>
   
    public let currentUser: UserType
    ...
}

public typealias User = Client.UserType
public typealias Channel = Client.ChannelType
```
Then developers can always use `User` and `Channel` without extra data generics.

Reply:
Yes, we should talk about this. The suggested solution is specified here https://github.com/GetStream/stream-chat-swift/blob/6288ba6e4a5b83d2df3764acf6bb74a3df4a999c/StreamChatClient_v3/ChatClient.swift#L9 . Take a look.

Reply:
They won't need to use these types with a generic. Please check my idea example above.